pkg/doc/verifiable/credential.go

```diff
@@ -276,20 +320,38 @@ func decodeIssuer(data []byte, credential *Credential) error {
 
 // NewCredential creates an instance of Verifiable Credential by reading a JSON document from bytes.
 // It also applies miscellaneous options like custom decoders or settings of schema validation.
-func NewCredential(dataJSON []byte, opts ...CredentialOpt) (*Credential, error) {
+func NewCredential(vcData []byte, opts ...CredentialOpt) (*Credential, error) {
 	// Apply options
 	crOpts := defaultCredentialOpts()
 	for _, opt := range opts {
 		opt(crOpts)
 	}
 
-	raw := &rawCredential{}
-	err := json.Unmarshal(dataJSON, &raw)
-	if err != nil {
-		return nil, fmt.Errorf("JSON unmarshalling of verifiable credential failed: %w", err)
+	var err error
+	var raw *rawCredential
+	var vcDataDecoded []byte
+	rawDecoding := true
+
+	if crOpts.issuerPublicKeyFetcher != nil {
+		var vcDataFromJwt []byte
+		if vcDataFromJwt, raw, err = decodeJws(vcData, crOpts.issuerPublicKeyFetcher); err != nil {
+			logger.Infof("cannot decode JWS: %w", err)
```

Comment:
I think we should simply abort and return the error at this point. One of the possible errors of `decodeJws()` results from validation of the signature. If that fails, we wouldn't want to continue.

Reply:
Also thought about it, but considered it as use case when e.g. a Holder can get Verifiable Credential with proof embedded into Verifiable Credential (**proof** property). From the other hand, we don't support embedded proofs for now.

@llorllale do you think we should abort and return the error at this point, right?

Reply:
I think we should, yes.

There are plenty of other possible error conditions in `decodeJws()` - I don't see why we need to distinguish between them on this line and attempt to decide whether it's worth continuing or not. Aborting is the simplest path forward.

Reply:
Deal!